sv_pipeline/genome/load_data_tests.py

```diff
@@ -0,0 +1,279 @@
+        self.vcf_file = tempfile.mkstemp(suffix='.vcf')[1]
+        with open(self.vcf_file, 'w') as f:
+            f.writelines('\n'.join(VCF_DATA))
+        hl.init()
+        self.mt = hl.import_vcf(self.vcf_file, reference_genome='GRCh38')
+
+    def tearDown(self):
+        hl.stop()
+        os.remove(self.vcf_file)
+
+    @mock.patch('sv_pipeline.genome.load_data.os')
+    @mock.patch('sv_pipeline.genome.load_data.hl')
+    @mock.patch('sv_pipeline.genome.load_data.logger')
+    def test_load_mt(self, mock_logger, mock_hl, mock_os):
+        mock_os.path.splitext.return_value = ['test_dataset/input_vcf', 'gz']
+        mock_os.path.isdir.return_value = True
+        _ = load_mt(TEST_INPUT_DATASET, '', False)
+        mock_hl.read_matrix_table.assert_called_with('test_dataset/input_vcf.mt')
+        mock_logger.info.assert_called_with('Use the existing MatrixTable file test_dataset/input_vcf.mt. '
+                                       'If the input VCF file has been changed, or you just want to re-import VCF,'
+                                       ' please add "--overwrite-matrixtable" command line option.')
+
+
+    @mock.patch('sv_pipeline.genome.load_data.get_sample_subset')
+    @mock.patch('sv_pipeline.genome.load_data.get_sample_remap')
+    @mock.patch('sv_pipeline.genome.load_data.logger')
+    def test_subset_mt(self, mock_logger, mock_get_remap, mock_get_sample):
+        mock_get_sample.return_value = {'SAMPLE-1', 'SAMPLE-3', 'SAMPLE-5', 'SAMPLE-6'}
+        with self.assertRaises(Exception) as e:
+            _ = subset_mt('test_guid', self.mt, skip_sample_subset=False, ignore_missing_samples=False)
+        self.assertEqual(str(e.exception), 'Missing the following 1 samples:\nSAMPLE-6')
+
+        mock_get_remap.return_value = {'SAMPLE-1': 'SAMPLE-1-REMAP'}
+        rows = subset_mt('test_guid', self.mt, skip_sample_subset=False, ignore_missing_samples=True)
+        calls = [
+            mock.call('Missing the following 1 samples:\nSAMPLE-6'),
+            mock.call('Subsetting to 4 samples (remapping 1 samples)'),
+        ]
+        mock_logger.info.assert_has_calls(calls)
+        mock_get_sample.assert_called_with('test_guid', 'WGS')
+        mock_get_remap.assert_called_with('test_guid', 'WGS')
+        self.assertEqual(rows.count(), 5)
+        self.assertEqual(set(rows.genotypes.map(lambda x: x.sample_id).take(1)[0]), {'SAMPLE-1-REMAP', 'SAMPLE-3', 'SAMPLE-5'})
+
+        mock_logger.reset_mock()
+        rows = subset_mt('test_guid', self.mt, skip_sample_subset=True)
+        self.assertEqual(rows.count(), 5)
+        self.assertEqual(set(rows.genotypes.map(lambda x: x.sample_id).take(1)[0]), {'SAMPLE-1', 'SAMPLE-2', 'SAMPLE-3', 'SAMPLE-4', 'SAMPLE-5'})
+
+    @mock.patch('sv_pipeline.genome.load_data.load_gencode')
+    def test_annotation(self, mock_load_gencode):
+        mock_load_gencode.return_value = GENE_ID_MAPPING
+        rows = subset_mt('test_guid', self.mt, skip_sample_subset=True)
```

Comment:
this test needs to test `annotate_fields` independently from `subset_mt`. I think this is actually an issue because of how you divide these functions - the `genotypes` annotation belongs in `annotate_fields` function (probably as a BASIC_FIELD) not as part of `subset_mt`. You can still do the filtration in `subset_mt` without needing to compute the genotypes field

Reply:
Hm... You are right. I will do pure subsetting in `subset_mt`.